woocommerce/woocommerce
# Prevent initializing coming soon feature if it's already initialized

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Addressed feedback in https://github.com/woocommerce/woocommerce/pull/50581#discussion_r1716419310

This PR adds a check to prevent initializing the coming soon feature if it's already initialized. This is to prevent the coming soon feature from being initialized twice when the user visits the setup wizard after it has already been initialized.

We could check if core profiler is completed or skipped to determine if the coming soon feature should be initialized. However, I realized we mark the core profiler as completed or skipped before calling the API for the coming soon. But I also realized we can check the other options to determine if the coming soon feature should be initialized.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new site with this branch
2. Skip or complete the core profiler
3. Go to /wp-admin/admin.php?page=wc-settings&tab=site-visibility
4. Set `Site visibility` to `Live`
5. Revisit the core profiler /wp-admin/admin.php?page=wc-admin&path=%2Fsetup-wizard
6. Skip or complete the core profiler
7. Confirm Site visibility remains `Live`



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
